Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM golang:1.19-alpine AS build
+WORKDIR /app
+COPY go.mod go.sum ./
+RUN go mod download
+COPY *.go ./
+COPY . ./
+RUN go build -o /gwiserver
+
+
+FROM alpine
```

Comment:
Can you describe why this pattern with builder image is used?

Reply:
Yes, this pattern creates two images and moves the binary from the first image to the second image.
The reason behind this pattern is to build the code first in an image that contains the Go compiler, by moving the code and dependencies to that image. 
And after the compilation, we move the binary from the first image to the second image, which we want to be smaller and safer for production usage.